docs/src/locators.md

```diff
@@ -71,115 +71,113 @@ await locator.ClickAsync();
 page.getByRole("button").click();
-
-// Works because we explicitly tell locator to pick the first element:
-page.getByRole("button").first().click(); // ⚠️ using first disables strictness
-
-// Works because count knows what to do with multiple matches:
-page.getByRole("button").count();
 ```
 
 ```csharp
-// Throws if there are several buttons in DOM:
 await page.GetByRole("button").ClickAsync();
+```
+
+On the other hand, Playwright understands when you perform a multiple-element operation,
+so the following call works perfectly fine when locator resolves to multiple elements.
 
-// Works because we explicitly tell locator to pick the first element:
-await page.GetByRole("button").First.ClickAsync(); // ⚠️ using First disables strictness
+```js
+await page.getByRole('button').count();
+```
 
-// Works because Count knows what to do with multiple matches:
+```python async
+await page.get_by_role("button").count()
+```
+
+```python sync
+page.get_by_role("button").count()
+```
+
+```java
+page.getByRole("button").count();
+```
+
+```csharp
 await page.GetByRole("button").CountAsync();
 ```
 
-:::caution
-Using [`method: Locator.first`], [`method: Locator.last`], and [`method: Locator.nth`] is discouraged since it disables the concept of strictness, and as your page changes, Playwright may click on an element you did not intend. It's better to make your locator more specific.
-:::
+You can explicitly opt-out from strictness check by telling Playwright which element to use when multiple element match, through [`method: Locator.first`], [`method: Locator.last`], and [`method: Locator.nth`]. These methods are **not recommended** because when your page changes, Playwright may click on an element you did not intend. Instead, follow best practices below to create a locator that uniquely identifies the target element.
 
 
 ## Locating elements
 
-Use [`method: Page.locator`] method to create a locator. This method takes a selector that describes how to find an element in the page. The choice of selectors determines the resiliency of the test when the underlying web page changes. To reduce the maintenance burden, we recommend prioritizing user-facing attributes and explicit contracts.
+Playwright comes with multiple built-in ways to create a locator. To make tests resilient, we recommend prioritizing user-facing attributes and explicit contracts, and provide dedicated methods for them, such as [`method: Page.getByText`]. It is often convenient to use the [code generator](./codegen.md) to generate a locator, and then edit it as you'd like.
+
+If you absolutely must use CSS or XPath locators, you can use [`method: Page.locator`] to create a locator that takes a [selector](./selectors.md) describing how to find an element in the page.
+
+Note that all methods that create a locator, such as [`method: Page.getByLabel`], are also available on the [Locator] and [FrameLocator] classes, so you can chain them and iteratively narrow down your locator.
```

Comment:
would love an example of this as I am not quite sure what it means. isn't getByLabel a locator so ... framelocator ? API docs has a nice example we could possibly use here:
```js
const locator = page.frameLocator('#my-frame').getByText('Submit');
await locator.click();
```